src/main/java/dev/blaauwendraad/masker/json/path/JsonPathParser.java

```diff
@@ -0,0 +1,102 @@
+package dev.blaauwendraad.masker.json.path;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parses a jsonpath literal into a {@link dev.blaauwendraad.masker.json.path.JsonPath} object.
+ * <p>
+ * The following features from jsonpath specification are not supported:
+ * <ul>
+ *  <li>Wildcard segments</li>
+ *  <li>Descendant segments</li>
+ *  <li>Wildcard selectors</li>
+ *  <li>Array slice selectors</li>
+ *  <li>Filter selectors</li>
+ *  <li>Function extensions</li>
+ *  <li>Escape characters</li>
+ * </ul>
+ */
+@ParametersAreNonnullByDefault
+public class JsonPathParser {
+
+    /**
+     * Parses an input literal into a {@link dev.blaauwendraad.masker.json.path.JsonPath} object.
+     * Throws {@link java.lang.IllegalArgumentException} when the input literal does not follow the jsonpath specification.
+     *
+     * @param literal a jsonpath literal to be parsed.
+     * @return a {@link dev.blaauwendraad.masker.json.path.JsonPath} object parsed from the literal.
+     */
+    @Nonnull
```

Comment:
Isn't it unnecessary with class level annotation?

Reply:
As far as I understand `ParametersAreNonnullByDefault` is not applicable to return values